packages/rum/src/domain/segmentCollection/segmentCollection.ts

```diff
@@ -135,24 +137,34 @@ export function doStartSegmentCollection(
       return
     }
 
+    const viewStats = getOrCreateViewStats(context.view.id)
```

Comment:
Might be just me, but it feels like with stats calculation `createNewSegment` does more that it should. Maybe it's just the naming or we can split the function?

Reply:
agree, could we move the calculation to the view stats file?